src/release/breaking-changes/navigator-and-page-api.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Navigator's page APIs breaking change
+description: >
+  Changes to Navigator's page APIs.
+---
+
+## Summary
+
+The [`Navigator`]'s page APIs are refactored so that they can integrate
+into Flutter's other pop mechanisms.
+
+## Context
+
+The `onPopPage` was added for cleaning up pages after a page is about to be popped.
+The developer can veto pop by returning false in the callback. This does not work well
+with other popping mechanisms in the framework, such as [`PopScope`] and cupertino back
+gestures.
+
+The page APIs needed to be refactored in order to integrate them together.
+
+
+## Description of change
+
+The `onPopPage` is replaced by `onDidRemovePage` and no longer had the ability to veto
+a pop. In the `onDidRemovePage`, one is only responsible for updating the [`pages`].
```

Comment:
**suggestion:** Write in active voice per [GSG](https://developers.google.com/style/voice). Google Style doesn't recommend using a property as a noun on its own. It should modify what it's changing.

```suggestion
The `onDidRemovePage` property replaces the `onPopPage` property.
It can no longer veto a pop.
In the `onDidRemovePage`, one is only responsible for updating the [`pages`].
```